BaseTools/Conf/build_rule.template

```diff
@@ -768,29 +768,16 @@
         GenFw -o $(OUTPUT_DIR)(+)$(MODULE_NAME)hii.rc -g $(MODULE_GUID) --hiibinpackage $(HII_BINARY_PACKAGES)
 
 # MU_CHANGE Start
-[Resource-File]
+[Resource-File.UEFI_HII]
     <InputFile>
         *.arc
 
-    <OutputFile.MSFT, OutputFile.INTEL, OutputFile.CLANGPDB>
-        $(OUTPUT_DIR)(+)$(MODULE_NAME)Resource.lib
-
-    <OutputFile.GCC>
-        $(OUTPUT_DIR)(+)$(MODULE_NAME)Resource.rc
-
-    <OutputFile.XCODE, OutputFile.RVCT>
-        $(OUTPUT_DIR)(+)$(MODULE_NAME)Resource.rc
-
-    <Command.MSFT, Command.INTEL, Command.CLANGPDB>
-        "$(RC)" /Fo${dst} ${src}
+    <OutputFile.MSFT, OutputFile.INTEL, OutputFile.GCC, OutputFile.CLANGPDB>
+        $(OUTPUT_DIR)(+)$(MODULE_NAME)Resource.hpk

```

Comment:
If we output HPK will this mean that version resource and hii resource are exclusive since the output file of HPK rule above is the same?  I like the reuse of HII but we should support a module that uses both.

Reply:
> If we output HPK will this mean that version resource and hii resource are exclusive since the output file of HPK rule 

No, because the content of all the different `.hpk` files will be added to the `$(MODULE_NAME)hii.rc`.

In other words, the `$(HII_BINARY_PACKAGES)` variable in the command of the `Hii-Binary-Package.UEFI_HII` rule gets expanded to all the `.hpk` files, so the final `.rsrc` section in the PE32+ binary will contain the data of all the `hpk` files.

For example, on my test machine this is the actual commands executed by the `Hii-Binary-Package.UEFI_HII` rule:

```
INFO - "GenFw" -o /home/javier/devel/mu_basecore/Build/MdeModule/DEBUG_GCC5/X64/MdeModulePkg/Application/HelloWorld/HelloWorld/OUTPUT/HelloWorldhii.rc -g 6987936E-ED34-44db-AE97-1FA5E4ED2116 --hiibinpackage /home/javier/devel/mu_basecore/Build/MdeModule/DEBUG_GCC5/X64/MdeModulePkg/Application/HelloWorld/HelloWorld/OUTPUT/HelloWorldResource.hpk /home/javier/devel/mu_basecore/Build/MdeModule/DEBUG_GCC5/X64/MdeModulePkg/Application/HelloWorld/HelloWorld/OUTPUT/HelloWorldStrDefs.hpk
INFO - "objcopy" -I binary -O elf64-x86-64 -B i386 --rename-section .data=.hii /home/javier/devel/mu_basecore/Build/MdeModule/DEBUG_GCC5/X64/MdeModulePkg/Application/HelloWorld/HelloWorld/OUTPUT/HelloWorldhii.rc /home/javier/devel/mu_basecore/Build/MdeModule/DEBUG_GCC5/X64/MdeModulePkg/Application/HelloWorld/HelloWorld/OUTPUT/HelloWorldhii.lib
```

Reply:
@spbrogan and thanks a lot for your feedback and review!

Reply:
so they all get merged into the one .rc?  

Have you tried running the version_info tool on the binary and see the results match expected? You will need  at least version 0.14.0 of edk2-pytool-extensions

`versioninfo_tool -d {/path/to/app.efi} {/path/to/output.JSON}`

https://github.com/tianocore/edk2-pytool-extensions/blob/master/docs/usability/using_versioninfo.md

Reply:
> so they all get merged into the one .rc?

Yes, my understanding is that this is already the case for other build rules that output a `.hpk` file.

> Have you tried running the version_info tool on the binary and see the results match expected? You will need at least version 0.14.0 of edk2-pytool-extensions

I was just testing with `strings -e l` and `objdump -s -j .rsrc` but I've tried now with that script and it fails...

```
$ strings -e l ./Build/MdeModule/DEBUG_GCC5/X64/HelloWorld.efi
UEFI Hello World!
VS_VERSION_INFO
VS_VERSION_INFO
StringFileInfo
040904b0
CompanyName
Example Company Name
OriginalFilename
HelloWorld.efi
VarFileInfo
Translation
English
.TH HelloWorld 0 "Displays a "UEFI Hello World!" string."
.SH NAME
HelloWorld application.

$ versioninfo_tool -d ./Build/MdeModule/DEBUG_GCC5/X64/HelloWorld.efi test.json
Could not find VS_FIXEDFILEINFO.
Traceback (most recent call last):
  File "/usr/local/bin/versioninfo_tool", line 33, in <module>
    sys.exit(load_entry_point('edk2-pytool-extensions==0.14.0', 'console_scripts', 'versioninfo_tool')())
  File "/usr/local/lib/python3.9/site-packages/edk2toolext/versioninfo/versioninfo_tool.py", line 105, in main
    if not decode_version_info_dump_json(args.input_file, args.output_file):
  File "/usr/local/lib/python3.9/site-packages/edk2toolext/versioninfo/versioninfo_tool.py", line 74, in decode_version_info_dump_json
    version_info = decode_version_info(input_file)
  File "/usr/local/lib/python3.9/site-packages/edk2toolext/versioninfo/versioninfo_tool.py", line 66, in decode_version_info
    return pe.get_version_dict()
  File "/usr/local/lib/python3.9/site-packages/edk2toolext/versioninfo/versioninfo_helper.py", line 395, in get_version_dict
    for fileinfo in self._pe.FileInfo:
AttributeError: 'PE' object has no attribute 'FileInfo'
```

To make sure that it wasn't a problem with the script, I just tried to load the binary and print the `PE` class `VS_FIXEDFILEINFO` attribute but that doesn't work either:

```
>>> import pefile
>>> path = "./Build/MdeModule/DEBUG_GCC5/X64/HelloWorld.efi"
>>> pe = pefile.PE(path)
>>> print(pe.VS_FIXEDFILEINFO)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'PE' object has no attribute 'VS_FIXEDFILEINFO'
>>>
```
I see that the same problem happens if I just take a binary that doesn't have a `.rsrc` section and attempt to embed a binary `.res` file with `objcopy HelloWorld.efi --add-section .rsrc=HelloWorldResource.rc`.

I will take a look tomorrow to understand what's missing or wrong with this approach.